### ISSUE TITLE
API refresh

### ISSUE THREAD

**Issue by mikabytes** [tidenjs/tiden#1]
- [x] Concept -> Stream
- [x] ~~Component -> Element~~ too much overlap with native Element, HTMLElement. Besides, keeping similarity to "web components" is desirable.
- [x] Confluence -> Merge (it's shorter to write, and well-known word)
- [x] Remove deprecated functions
- [x] Add decorators for Stream and Nano (for future debugging insight implementation)
- [x] New folder hierarchy: explicit folders for element/nano/stream/page. Remove artificial constraint of namespace/module. This will allow user to form their own structure.
- [x] Disallow arrays to be used as values in 'connect'
- [x] ~~Deprecate objects used as values~~ this will still be needed, so we can't remove it
- [x] Merges that return `undefined` to be treated as undefined (i.e. there is no definition of data). Remove constraints parameter
- [x] Navigate to use `id` when available, and fallback to `pageId`

**Comment by mikabytes** [comment]
Related to folder hierarchy, we might want to rename the `src` folder. According to [some](https://gist.github.com/tracker1/59f2c13044315f88bee9) [resources](https://stackoverflow.com/questions/38426078/why-use-lib-vs-src-directory-names-in-javascript-which-is-standard) `src` communicates that this is code that needs to be compiled before it can be used. This is most definitely not the case with Tiden apps as we do not compile code.

In NodeJS `lib` communicates code that is ready to be used by an external party. While this is more appropriate, it's not really an ideal fit either. Tiden nanos and streams _may_ be exported, but in the normal case they are an integral part of the app that we're building.

We could adopt the [convention set by rails](https://www.tutorialspoint.com/ruby-on-rails/rails-directory-structure.htm). This is currently my favorite. The "app" building blocks goes in a folder called... `app`. Simple enough, very clear.

If we decide to go this way, then the root folder structure might look something like this:

```
public/
   ... static files
lib/
  ... any helpers or general purpose libraries goes here
app/
  streams/ (these core-level streams are provided by stdlib #7 )
    url.js
    audio.js
    navigator.js
  someModule/
    streams/
      ...
     nanos/
       ...
     pages/
        ...
```

**Comment by mikabytes** [comment]
We might want to consider renaming Element to CustomElement not to conflict with native web [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element)

**Comment by mikabytes** [comment]
Merges that return `undefined` was fixed in f0a6c1c7
